papermill/api.py

```diff
@@ -140,43 +161,51 @@ def record_value(name, value):
     display({RECORD_OUTPUT_TYPE: {name: value}}, raw=True)
 
 
-def fetch_record(cell, name):
+def fetch_notebook_data(notebook):
     """
-    Fetches a recorded value from the cell under inspection.
+    Returns a dict of all the recorded data from the execution of the notebook.
 
     Args:
-        cell (nbformat.NotebookNode): The cell to pull the value from.
-        name (str): Then name of the value.
+        notebook (nbformat.NotebookNode or str): Notebook or path to notebook under inspection.
 
     Returns:
-        The value retrieved from the cell's output at the specified name.
-
+        Dictionary of parameters for the notebook and the recorded notebook data.
     """
-    value = None
-    for output in cell.get('outputs', []):
-        if 'data' in output and RECORD_OUTPUT_TYPE in output['data']:
-            if name in output['data'][RECORD_OUTPUT_TYPE]:
-                value = output['data'][RECORD_OUTPUT_TYPE][name]
-    return value
+    if isinstance(notebook, basestring):
+        # Pull from path if notebook is a string.
+        nb = read_notebook(notebook)
+    else:
+        nb = notebook
 
+    ret = {}
+    ret.update(nb.metadata.papermill['parameters'])
+    for cell in nb.cells:
+        for output in cell.get('outputs', []):
+            if 'data' in output and RECORD_OUTPUT_TYPE in output['data']:
+                ret.update(output['data'][RECORD_OUTPUT_TYPE])
+    return ret
 
-def _split_bucket_and_prefix(path):
+
+def fetch_notebooks_dataframe(notebooks):
     """
-    splits an s3 path into bucket and prefix, like os.path.split, only it can only be used once
-    ie s3://foo/bar/baz would return ['foo','bar/baz']
-    note: because a trailing / is significant in s3, it will not be stripped,
-        ie s3://foo/bar/baz/ will return ['foo','bar/baz/']
+    Returns a dataframe where each row is a notebook in a directory and each column is a parameter
+    or recorded data value from the notebooks.
+
+    Args:
+        notebooks (list): List of NotebookNode or paths to the notebook directory.
+
+    Returns:
+        pd.DataFrame containing the data from the notebooks in the directory.
     """
-    if not path.startswith('s3://'):
-        raise ValueError('path must start with s3://')
-    path = path[5:]
-    if path:
-        if '/' in path:
-            return path.split('/', 1)
-        else:
-            return [path, '']
+    if isinstance(notebooks, basestring):
```

Comment:
Use the string handling for both Python 2 and Python 3 from the other PR so this supports both. I'll help fix the build in the other PR in the morning.